Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -240,51 +354,63 @@ public struct HTTPHeaders: CustomStringConvertible {
     /// - Parameter name: The header field name whose values are to be retrieved.
     /// - Returns: A list of the values for that header field name.
     public subscript(name: String) -> [String] {
-        if let result = storage[name.lowercased()] {
-            return result.map { tuple in tuple.1 }
+        guard !self.headers.isEmpty else {
+            return []
         }
-        return []
-    }
 
-    /// Serializes this HTTP header block to bytes suitable for writing to the wire.
-    ///
-    /// - Parameter buffer: A buffer to write the serialized bytes into. Will increment
-    ///     the writer index of this buffer.
-    func write(buffer: inout ByteBuffer) {
-        for (key, values) in storage {
-            if key != "set-cookie" {
-                writeListHeaderValues(buffer: &buffer, key: key, values: values)
-            } else {
-                writeSequentialHeaderValues(buffer: &buffer, key: key, values: values)
+        let utf8 = name.utf8
+        var array: [String] = []
+        for header in headers {
+            if self.buffer.equalCaseInsensitiveASCII(view: utf8, at: header.name) {
+                array.append(string(idx: header.value))
```

Comment:
Can we use the explicit `self` here? Took me a while to work out where this `String` constructor came from. :wink: